File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
# def solution(arr):
#     answer = []
#     for i in arr:
#         if i not in answer:
#             answer.append(i)
#         elif answer[-1] != i:
#             answer.append(i)
#     return answer

def solution(arr):
    answer = []
    for i in arr:
        if answer[-1:] != [i]:
            answer.append(i)
    return answer

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        Deque<Integer> que = new ArrayDeque<>();
        for (int a : arr) {
            if (que.isEmpty() || que.peekLast() != a) {
                que.offer(a);
            } 
        }
        int[] answer = new int[que.size()];
        int idx = 0;
        for (Integer q : que) {
            answer[idx++] = q;
        }
        return answer;
    }
}